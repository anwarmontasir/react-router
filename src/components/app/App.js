import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import Header from './Header';
import Search from '../search/Search';
// import MovieDetail from '../movies/MovieDetail';

import './App.css';

export default class App extends Component {
  render() {
    
    return (
      <Router>
        <div>
          <Header/>
          <main>
            <Switch>
              <Route exact path="/" component={Home}/>
              <Route path="/search" component={Search}/>
              <Redirect to="/"/>
            </Switch>
          </main>
          <footer>&copy; 2018 this app.</footer>
        </div>
      </Router>
    );
  }
}

class Home extends Component {
  render() {
    return <div>I am Home!</div>;
  }
}