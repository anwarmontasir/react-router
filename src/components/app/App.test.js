import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from './App';

describe('App', () => {
  test('no home test', () => {
    const wrapper = shallow(<App/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('with home', () => {
    const wrapper = shallow(<App home={true}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});