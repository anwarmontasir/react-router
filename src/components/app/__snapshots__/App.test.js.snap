// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App no home test 1`] = `
<BrowserRouter>
  <div>
    <Header />
    <main>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Redirect
          push={false}
          to="/"
        />
      </Switch>
    </main>
    <footer>
      © 2018 this app.
    </footer>
  </div>
</BrowserRouter>
`;

exports[`App with home 1`] = `
<BrowserRouter>
  <div>
    <Header />
    <main>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Redirect
          push={false}
          to="/"
        />
      </Switch>
    </main>
    <footer>
      © 2018 this app.
    </footer>
  </div>
</BrowserRouter>
`;
